src/blobs.rs

```diff
@@ -7,38 +7,38 @@
 
 //! Iterator and utilities for `fileformat::Blob`.
 
-use blocks;
-use fileformat::Blob;
-use objects::OsmObj;
-use osmformat::PrimitiveBlock;
 use std::iter;
-use Result;
 
-rental! {
-    mod rent {
-        use osmformat::PrimitiveBlock;
-        use blocks;
-        #[rental]
-        pub struct OsmObjs {
-            block: Box<PrimitiveBlock>,
-            objs: blocks::OsmObjs<'block>,
-        }
+use crate::blocks;
+use crate::blocks::OsmObjs as OsmBlockObjs;
+use crate::fileformat::Blob;
+use crate::objects::OsmObj;
+use crate::osmformat::PrimitiveBlock;
+use crate::Result;
+
+self_cell!(
+    struct OsmBlobObjs {
+        #[from_fn]
+        owner: PrimitiveBlock,
+
+        #[covariant]
+        dependent: OsmBlockObjs,
```

Comment:
Can we `blocks::OsmObjs` as before or is it a limitation of your crate?

Reply:
Indeed that's a limitation of this crate, passing a type without the lifetime can be done in a couple ways but without access to `concat_ident` `ident` is the only way I found that meets my requirements.